src/shims/unix/linux/foreign_items.rs

```diff
@@ -143,6 +144,21 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     id if id == sys_futex => {
                         futex(this, &args[1..], dest)?;
                     }
+                    id if id == sys_eventfd2 => {
+                        if args.len() < 3 {
+                            throw_ub_format!(
+                                "incorrect number of arguments for `eventfd2` syscall: got {}, expected at least 3",
+                                args.len()
+                            );
+                        }
+
+                        let result = this.eventfd(&args[1], &args[2])?;
+                        // TODO what's a cleaner way to go from a Scalar i32 to Scalar integer?
```

Comment:
I think `to_int` is what you're looking for? https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_middle/mir/interpret/value.rs.html#403-406

Reply:
> I think `to_int` is what you're looking for? https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_middle/mir/interpret/value.rs.html#403-406

Thank you. Do you know what value for the Size type to pass in?

Reply:
IMO going via a host `i32` is completely fine in this case.

The alternative would be to call `int_to_int_or_float`.

Reply:
> IMO going via a host `i32` is completely fine in this case.

Sorry. I'm not understanding. When I first plumbed this together, I wasn't trying to change the Scalar::from_i32(fd_value) that the eventfd function returns and the Miri runtime was reporting  a write with uninitialized data. It took a lot of trial and error before hitting on the idea that the four bytes being created by the eventfd function weren't enough to write the result to an eight byte result that syscall is designed to return.

But is there something else you mean by host `i32`?

Also, I don't see how to convert the type I have to one that `this.int_to_int_or_float` would accept. I have a `Scalar<Provenance>` and the method takes an `ImmTy<'_, machine::Provenance>`.

Reply:
I don't have the answer to the question, but I wonder if it helps to do this:

```rust
let result = this.eventfd(&args[1], &args[2])?;
this.write_int(result.to_i32()?, dest)?;
```

Reply:
Going via host `i32` is what you are doing by calling `to_i32()`.

And yes, the proposal @tiif makes looks great, and it's more compact than what the PR does, so let's do that. :)

Reply:
But that doesn't compile.
```
error[E0277]: the trait bound `rustc_const_eval::interpret::Scalar<machine::Provenance>: std::convert::From<i32>` is not satisfied
   --> src/shims/unix/linux/foreign_items.rs:161:43
    |
161 |                         this.write_scalar(result.to_i32()?, dest)?;
    |                              ------------ ^^^^^^^^^^^^^^^^ the trait `std::convert::From<i32>` is not implemented for `rustc_const_eval::interpret::Scalar<machine::Provenance>`, which is required by `i32: std::convert::Into<rustc_const_eval::interpret::Scalar<machine::Provenance>>`
    |                              |
    |                              required by a bound introduced by this call
    |
    = help: the following other types implement trait `std::convert::From<T>`:
              `rustc_const_eval::interpret::Scalar<Prov>` implements `std::convert::From<rustc_apfloat::ieee::IeeeFloat<rustc_apfloat::ieee::DoubleS>>`
              `rustc_const_eval::interpret::Scalar<Prov>` implements `std::convert::From<rustc_apfloat::ieee::IeeeFloat<rustc_apfloat::ieee::HalfS>>`
              `rustc_const_eval::interpret::Scalar<Prov>` implements `std::convert::From<rustc_apfloat::ieee::IeeeFloat<rustc_apfloat::ieee::QuadS>>`
              `rustc_const_eval::interpret::Scalar<Prov>` implements `std::convert::From<rustc_apfloat::ieee::IeeeFloat<rustc_apfloat::ieee::SingleS>>`
              `rustc_const_eval::interpret::Scalar<Prov>` implements `std::convert::From<rustc_middle::ty::ScalarInt>`
    = note: required for `i32` to implement `std::convert::Into<rustc_const_eval::interpret::Scalar<machine::Provenance>>`
note: required by a bound in `rustc_const_eval::interpret::place::<impl rustc_const_eval::interpret::InterpCx<'tcx, M>>::write_scalar`
```

Reply:
I'm removing the TODO comment.

Reply:
write_int works, which was suggested above

Reply:
Thanks for the help!